javagen/src/main/resources/MultipartFormDataHelper.java

```diff
@@ -0,0 +1,123 @@
+import com.azure.core.http.HttpHeaderName;
+import com.azure.core.http.rest.RequestOptions;
+import com.azure.core.util.BinaryData;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.io.SequenceInputStream;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.UUID;
+
+// DO NOT modify this helper class
+
+public final class MultipartFormDataHelper {
+    /**
+     * Line separator for the multipart HTTP request.
+     */
+    private static final String CRLF = "\r\n";
+
+    /**
+     * Value to be used as part of the divider for the multipart requests.
+     */
+    private final String boundary;
+
+    /**
+     * The actual part separator in the request. This is obtained by prepending "--" to the "boundary".
+     */
+    private final String partSeparator;
+
+    /**
+     * The marker for the ending of a multipart request. This is obtained by post-pending "--" to the "partSeparator".
+     */
+    private final String endMarker;
+
+    /**
+     * Charset used for encoding the multipart HTTP request.
+     */
+    private final Charset encoderCharset = StandardCharsets.UTF_8;
+
+    private InputStream requestDataStream = new ByteArrayInputStream(new byte[0]);
+    private long requestLength = 0;
+
+    private RequestOptions requestOptions;
+    private BinaryData requestBody;
+
+    /**
+     * Default constructor used in the code. The boundary is a random value.
+     */
+    public MultipartFormDataHelper(RequestOptions requestOptions) {
+        this(requestOptions, UUID.randomUUID().toString().substring(0, 16));
+    }
+
+    private MultipartFormDataHelper(RequestOptions requestOptions, String boundary) {
```

Comment:
Since this is private, we don't really need this overload. We can move the logic to the public ctor.

Reply:
If we want to make `boundary` configurable, then we can make this ctor public.

Reply:
Sure. Merged into the public ctor.